ddtrace/filters.py

```diff
@@ -75,6 +75,11 @@ def process_trace(self, trace):
         return trace
 
 
+@removals.removed_class(
+    "TraceCiVisibilityFilter",
+    message="TraceCiVisibilityFilter is deprecated and will be removed from the public API.",
+    removal_version="2.0.0",
+)
 class TraceCiVisibilityFilter(TraceFilter):
```

Comment:
I am not a big fan of introducing two definitions of `TraceCiVisibilityFilter` even if one has a deprecation warning. 

suggestion: Can we make `ddtrace.ci_visibility._filters.TraceCiVisibilityFilter` backwards compatible and import it in this module. With this approach we might to update the `removals.removed_class` annotation.

Reply:
`TraceCiVisibilityFilter` should never have been introduced into the public API, as there's no use case for anyone to use it outside of CI Visibility, and the new one has introduced breaking changes (to fit CI-visibility specific logic). 

Essentially with this refactor, `ddtrace.ci_visibility._filters.TraceCiVisibilityFilter` is not backwards compatible with the existing one, so I don't see a problem with keeping the public API unchanged, but deprecating it for removal in 2.0.